iml-agent/src/action_plugins/kernel_module.rs

```diff
@@ -0,0 +1,9 @@
+use crate::agent_error::ImlAgentError;
+
+use kmod;
+
+pub async fn loaded(module: String) -> Result<bool, ImlAgentError> {
+    let ctx = kmod::Context::new()?;
+
+    Ok(ctx.modules_loaded()?.find(|m| m.name() == module).is_some())
```

Comment:
Since this is not async by itself, this call should be inside a blocking thread.

ref: 

https://github.com/whamcloud/integrated-manager-for-lustre/blob/203a2d67cdb8bdb0ece520f01e1997ac451e128e/iml-agent/src/action_plugins/stratagem/action_purge.rs#L24-L26

Reply:
Right, I've been thinking about this too